Anagram.java

```diff
@@ -28,22 +30,65 @@ public static void main(String args[]) {
 
 	// Returns true if the two given strings are anagrams, false otherwise.
 	public static boolean isAnagram(String str1, String str2) {
-		// Replace the following statement with your code
-		return false;
+		String str1Check = preProcess(str1);
+		String str2Check = preProcess(str2);
+		int checker = 0;
+		for (int i = 0; i < str2Check.length() ; i++){
+			for (int j = 0; j < str1Check.length() ; j++){
+				if (str2Check.charAt(i) == str1Check.charAt(j)){
+					checker++;
+				}
+			}
+			if (checker == 0){
+				return  false;
+			}
+			checker = 0;
+		}
+		return true;
 	}
 	   
 	// Returns a preprocessed version of the given string: all the letter characters are converted
 	// to lower-case, and all the other characters are deleted, except for spaces, which are left
 	// as is. For example, the string "What? No way!" becomes "whatnoway"
 	public static String preProcess(String str) {
-		// Replace the following statement with your code
-		return "";
+		String str1 = str;
+		String newStr = "";
+		for ( int i = 0; i < str1.length() ; i++){
+			char x = str1.charAt(i);
+			if ( (x >= 'A') && ( x <= 'Z') ){
+				x = (char)(str1.charAt(i) + 32);
+				newStr = newStr + x;
+			} else if ( (x >= 'a') && (x <= 'z')) newStr = newStr + x;
+			  else if (( x == ' ')) newStr = newStr + " ";	
+		}
```

Comment:
1. really hard to read.. 
2. export to functions.. 
3. could structure better.